### ISSUE TITLE
How to include interpolation service for Berlin, Germany?

### ISSUE THREAD

**Issue by jengeb** [pelias/interpolation#112]
Dear Pelias maintainers, 

I use the [dockerfiles repository](https://github.com/pelias/dockerfiles) and would actually like to set up a local instance of Pelias with the data of Berlin/Germany. I've already searched for possibilities to adjust the data pathes and files in a config file called [pelias.json](https://github.com/pelias/dockerfiles/blob/master/pelias.json). 
The interpolation part for Portland, Oregon is the following:
``` 
"interpolation": {
  "download": {
    "tiger": {
      "datapath": "/data/tiger",
      "states": [
        {
          "state_code": 41
        }
      ]
    }
  }
}
```
Until now I've found no way to configure this part for Berlin. 

But I executed the following code of your [readme](https://github.com/pelias/interpolation/blob/master/readme.md) and got this error message (I also tried using `sudo`):

```
$ curl -s http://missinglink.files.s3.amazonaws.com/berlin.gz | gzip -d > /tmp/data/berlin.0sv
$ bash: /tmp/data/berlin.0sv: Permission denied
```

Do you know a way to configure the interpolation for Berlin? I would be very happy if anyone could help me.

**Comment by otbutz** [comment]
Please check the permissions of `/tmp/data`:
```sh
ls -lah /tmp
```

Your user should have write permissions. This most likely the reason why you receive the `Permission denied` error.

**Comment by orangejulius** [comment]
Hi @jengeb,
[TIGER](https://en.wikipedia.org/wiki/Topologically_Integrated_Geographic_Encoding_and_Referencing) is a dataset that is only relevant for the US. I'm not 100% sure but you might be able to remove the `"tiger"` section completely. @missinglink can confirm.

**Comment by jengeb** [comment]
Thank you very much for your answers. I am still a bit confused how to apply interpolation for a non US region or city. Starting with the [pelias.json](https://github.com/pelias/dockerfiles/blob/master/pelias.json) config file, I searched for and adjusted the imported datasets for openaddresses, polyline etc. I thought I could do the same with interpolation.
I read the [How to guide](https://github.com/pelias/dockerfiles/blob/master/how_to_guide.pdf), but only found instructions for using tiger and US data. Is there another guide explaining how to use interpolation with non US data?

**Comment by otbutz** [comment]
>  Is there another guide explaining how to use interpolation with non US data?

Here you go: https://github.com/pelias/interpolation#building-the-databases

tl;dr:
* create `street.db` from your polyline data
* use your `street.db` and your openaddresses data to create `address.db`
* calculate vertices using both `address.db` and `street.db`
* start your polyline service

**Comment by missinglink** [comment]
Hi @jengeb sorry for the late reply on this one.

Julian is correct in saying that the TIGER block range data comes from the US CENSUS and so is only relevant in the USA.

I'm the author of the interpolation library and I'm also based in Berlin, so you'll actually find a few test cases for Berlin in the test directory.

I saw you opened a few other issues and PRs across the Pelias org, if you're interested in contributing more to the project I'd be happy to meet you for a coffee to talk about your experience of installing/running the code and the requirements for your project.

**Comment by jengeb** [comment]
Hi @missinglink,

Thanks for your response. For my project I set up a local instance of Pelias only including data for Berlin and Brandenburg. The goal ist to have a self hosted version of Pelias running as a geocoder for a map application. This app will only use the [endpoints](https://github.com/pelias/dockerfiles#api) for searching and autocompleting addresses. 

After getting docker and the following command running (see #117 ) I could [build the databases](https://github.com/pelias/interpolation#building-the-databases) as described. 

```bash
$ cat /data/brandenburg.polylines | docker run -i -v /data:/data pelias/interpolation polyline /data/street.db
```

I am using `pelias.json` of the [pelias/dockerfiles](https://github.com/pelias/dockerfiles) repo as a configuration file. I still wonder if it is possible to just configure the interpollation part of `pelias.json` for my needs (Berlin, Brandenburg datasets) - all examples I could find use tiger and US datasets. Or would I leave the interpolation part of `pelias.json` empty after running the steps to build the databases? Meaning do these steps import data and populate the elastic (or the db where all the other datasets specified in `pelias.json` are stored) with information to use interpolation?

Best,
Jenny

**Comment by nilsnolde** [comment]
Hi @jengeb,
I'm currently wondering the same thing. Actually the exact same use case: Berlin, docker-compose images, wanting to use oa/osm interpolation. It's testing really, will set up a proper Pelias instance once the full capabilities become clear to me. Still bugging me though.. 

Did you get interpolation to work within the docker-compose pelias.json?

**Comment by missinglink** [comment]
here's some examples of what the service can do:

- https://interpolation.synergy.io/search/geojson?lat=52.500545&lon=13.313151&number=140&street=kurfursten%20damm
- https://interpolation.synergy.io/street/near/geojson?lat=52.498010244522554&lon=13.295942957047375

**Comment by jengeb** [comment]
@missinglink: Thanks a lot for the detailed explanation. 

@nilsnolde: I've left the [interpolation part](https://github.com/pelias/dockerfiles/blob/f6aa6beef1cb6f5e11544c66e754b79169bacf04/pelias.json#L67-L78) of the imports section in `pelias.json` out and built the index for interpolation as described [here](https://github.com/pelias/interpolation#building-the-databases).
